src/js/components/DataTable/index.d.ts

```diff
@@ -99,7 +99,13 @@ export interface DataTableProps<TRowType = any> {
       pad?: PadType;
     };
   };
-  rowDetails?: (row: TRowType) => React.ReactNode;
+  rowDetails?:
+    | ((row: TRowType) => React.ReactNode)
+    | {
+        render: (row: TRowType) => React.ReactNode;
+        expand?: Array<string>;
```

Comment:
should we also add number for the case if they use index

Reply:
We certainly can although we currently don't for `groupBy.expand`, `expandable` or `select` which are arrays of primary keys